### DISCUSSION TITLE
Deep API Review

### DISCUSSION THREAD

**Discussion by robloo** [AvaloniaUI/Avalonia#10665] [General]
Avalonia v11 makes an attempt to clean-up and standardize the API -- in many ways closer to WPF or WinUI. However, there is no deep review done and everything is just fixed as it is spotted. This means there are still things in v11 that probably should change and are just slipping through the cracks for now.

For the next major release v12 I would propose some infrastructure to, at least on the control level, diff the API between WPF/WinUI and Avalonia so we can really see the differences. Then have a much more direct attempt at closing the gap or clearly evolving to better concepts when necessary.

Spot checking API's is needed due to time constraints and resources right now but it just isn't robust. A more detailed design/API review should really be attempted at some point in the future.

**Comment by maxkatz6** [comment]
Note, this PR was created specifically to review the API https://github.com/AvaloniaUI/Avalonia/pull/10051
But it was mostly used to detect if anything should be made internal instead of correcting names and comparing to WPF/UWP.

**Reply by robloo** [reply]
Oh, nice! I didn't look at that PR carefully before. That type of system can definitely power a comparison app.

I was also thinking there should be a UI for it for devs to look at specific types side-by-side. Shouldn't be too difficult and would also potentially be more generally useful in other projects. I might look at that myself at some point.
